### PR TITLE
Removed timeout from subscribable next method.

### PR DESCRIPTION
**Pull request by evrii**
Description:

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

**Comment by dmohs** [comment]
This may have been the behavior that CH was concerned about earlier today:
```javascript
> ibreak = () => { throw 'I broke.' }
[Function: ibreak]
> ibreak()
Uncaught 'I broke.'
> setTimeout(ibreak, 1000)
Timeout {...}
> Uncaught 'I broke.'
> icatch = f => { try { f() } catch (e) { console.error('I caught:', e) }}
[Function: icatch]
> icatch(ibreak)
I caught: I broke.
undefined
> icatch(() => setTimeout(ibreak, 1000))
undefined
> Uncaught 'I broke.'
```
The above REPL session shows that a function placed at the end of the event loop (through `setTimeout`, for example) executes in a different context, which is why the `try` doesn't catch the error in that case.

In the original code, the `try` surrounded the subscriber function, which is a reasonable way to go, so I don't anticipate a change in behavior.

File: ui/src/app/utils/subscribable.ts [hunk 1]
```diff
@@ -45,18 +45,14 @@ export function subscribable<T>(): Subscribable<T> {
...
-            }
-          }, 0),
-        subscribers
-      );
+      fp.forEach((fn: subscriber) => {
+        try {
+          fn(newValue, oldValue);
+        } catch (e) {
+          // Ignore the error - ideally the supplied fn should
+          // supply its error handling
```
Comment (dmohs) [inline ui/src/app/utils/subscribable.ts:53]: I really like the narrow scope of this PR, so I would not change it, but I'd very much like to see (perhaps in a follow-up PR) what would happen if we:
```suggestion
          console.error('Uncaught error in subscriber function:', e)
```

**Comment by calbach** [comment]
I prefer the existing semantics for isolation purposes. As a caller of `atom.next()`, I think it's surprising to have all subscribers execute synchronously within that stack. This is not how most reliable event stores I've worked with typically function. The concrete impact of this is potentially creating implicit dependencies within subscribers on synchronous update behavior, or potential stack overflow on dependency chains of subscribers which are themselves invoking `next()`.

Could we add to the PR description the rationale for this change? This has always been the behavior, and we have many tests that are successfully using stores; while it may be the proximal cause of the test behavior we're seeing, I'm confident there are alternative solutions.

**Comment by calbach** [comment]
FYI: for reference, see original comment thread on this subject: https://github.com/all-of-us/workbench/pull/3665#discussion_r439691648

**Pull request closed without merging**